### ISSUE TITLE
ReadFCStd node erroneoud model.

### ISSUE THREAD

**Issue by macdroid53** [nortikin/sverchok#4921]
## Problem statement

When reading FreeCAD model the model is not read correctly.

## Steps to reproduce

1. Create and save model in FreeCAD
2. Use Read FCStd node to read the model
3. Model is wrong in Blender. Checks and test of the file in FreeCAD (for example it exports as a step file fine.)

## Expected result

Same  in Blender

## Actual result

What did you actually get?

## Sverchok version

v1.2.0-alpha
Blender 3.5.1

FreeCAD and Blend file:
[Archive.zip](https://github.com/nortikin/sverchok/files/11738186/Archive.zip)
![Screenshot_20230613_145711](https://github.com/nortikin/sverchok/assets/6831752/9c202ea8-e4c0-46b0-b11a-db2303996f1c)
![Screenshot_20230613_145741](https://github.com/nortikin/sverchok/assets/6831752/b6ee8eb1-f463-450e-8767-ef7da370d1dd)

**Comment by macdroid53** [comment]
> are you reading in the _body_? are you using `FCStd read Mod` node? ![image](https://user-images.githubusercontent.com/619340/245857280-2bbb5000-16b9-440a-b105-fbe6d7b004e3.png)
> 
>     * Blender 3.2
> 
>     * sv  'version': (1, 1, 0)

I'm using solid to mesh. If you bake in your example, you get a awful mesh. Using the Solid to mesh, with Trivial set, the mesh is very good. (works fine for other models I've used)

**Comment by zeffii** [comment]
I recall needing to implement the "Lenient" mesher algorithm for reasons related to this, a quick issue search might explain better than I can remember. 

![image](https://github.com/nortikin/sverchok/assets/619340/974485d0-99e2-42dc-bdbb-d970ccf05804)


maybe a face dissolver after the solid_to_mesh?

**Comment by macdroid53** [comment]
> I recall needing to implement the "Lenient" mesher algorithm for reasons related to this, a quick issue search might explain better than I can remember.
> 
>
> 
> maybe a face dissolver after the solid_to_mesh?
> 
>
> 
> we can't make polygons with holes.
Hmm...Can't seem to reproduce what you did there:
![Screenshot_20230615_113740](https://github.com/nortikin/sverchok/assets/6831752/0aae9b0e-4285-48ce-8e8f-78435f48cc18)

My Limited dissolve node is not the same:
![Screenshot_20230615_114011](https://github.com/nortikin/sverchok/assets/6831752/2f3fd83d-7804-4cef-8782-a4df775de7cf)

Also, not sure what you mean about making polygons with holes? As in holes in an ngon?

**Comment by zeffii** [comment]
the Limited Dissolve node looks the same as mine, except I manually changed the `angle` value to 0.001 and left the input field active to show that fact ( as soon as you leave/accept the slider value then the angle value shows a rounded value of 0.0 , and the result of 0.0 is not the same as 0.001 )

> Also, not sure what you mean about making polygons with holes? As in holes in an ngon?

yeah, i might misunderstanding your question but: the algorithm that converts from the `mathematical concept of a surface (which can include holes) ` to something that can be represented in Blender/Sverchok` is going to have to decompose it to a structure compatible with a polygonal mesh ( which doesn't support holes inside polygons )

**Comment by macdroid53** [comment]
> the Limited Dissolve node looks the same as mine, except I manually changed the `angle` value to 0.001 and left the input field active to show that fact ( as soon as you leave/accept the slider value then the angle value shows a rounded value of 0.0 , and the result of 0.0 is not the same as 0.001 )

Hmm...but, I don't get the result you show.
Here the Solid View is correct, but the mesh view fails. (If I use a Mesh Viewer node it produces what is expected.)
![Screenshot_20230616_070500](https://github.com/nortikin/sverchok/assets/6831752/d45beb2b-7799-44e9-92c9-3fe9c00ec872)
![Screenshot_20230616_071204](https://github.com/nortikin/sverchok/assets/6831752/30b40b89-6a8f-4766-a1d5-d495cfed9aaf)

**Comment by zeffii** [comment]
Yeah i see in your image..  and does the Terminal  display any more debug information about that error? 

I might have the chance to look at this tomorrow, no promises.

**Comment by rendetto** [comment]
The subject of this issue seems misleading because the problem is not about the ReadFCStd node.

@macdroid53 The model you're getting is fine in terms of algorithm functionality. You just need to enter better precision numbers. As an example with the Standard algorithm:

![Solid_To_Mesh](https://github.com/nortikin/sverchok/assets/66558924/0463fc75-3834-450f-9f1d-c28813fd4964)

As for the Lenient mesher... @zeffii can you give more info about it. It seems there is a bug with the UI not refreshing. So every time I change the tessellation method in the "Solid to Mesh" and get back to Lenient the previous UI stays:

![Lenient_UI_Issue](https://github.com/nortikin/sverchok/assets/66558924/8cb0468f-2389-4d3b-a49b-e14a7b8f1d79)

So it seems I never really tested the Lenient method because of the UI problem.

**Comment by zeffii** [comment]
> So every time I change the tessellation method in the "Solid to Mesh" and get back to Lenient the previous UI stays

I see what you mean, the Lenient mode indeed doesn't seem to correctly adjust the UI.

**Comment by luzpaz** [comment]
Any progress on this issue ?

**Comment by zeffii** [comment]
> > So every time I change the tessellation method in the "Solid to Mesh" and get back to Lenient the previous UI stays
> 
> I see what you mean, the Lenient mode indeed doesn't seem to correctly adjust the UI. The mode accepts no parameters, so it should not even show any other input sockets beyond `Solid` .. it's missing something like
> 
> 
https://github.com/nortikin/sverchok/commit/48995d30c073929fa87d0ffe2e3d17d73a4c30ae

i committed this, I thought someone else might have done it in the interim. To be clear this only fixes the mode change/syncing issue..
